lib/src/base_client.dart

```diff
@@ -23,15 +21,15 @@ abstract class BaseClient implements Client {
   /// can be a [Uri] or a [String].
   ///
   /// For more fine-grained control over the request, use [send] instead.
-  Future<Response> head(url, {Map<String, String> headers}) =>
-    _sendUnstreamed("HEAD", url, headers);
+  FutureOr<Response> head(url, {Map<String, String> headers}) =>
```

Comment:
Same question here: why `FutureOr`? This is never going to return a `Response` synchronously.